src/activations.jl

```diff
@@ -822,21 +840,23 @@ for (f, df) in UNARY_ACTS
     end
 end
 
+NO_ACT_GRAD = ChainRulesCore.@not_implemented "for simplicitly NNlib assumes the 2nd argument of this activation function is a constant"
+
 BINARY_ACTS = [ # f, df1, df2
-    (:elu, :(deriv_elu(Ω, x2)), :(NoTangent())), # TODO use real deriv instead of DNE
-    ]
+    (:elu, :(deriv_elu(Ω, x2)), NO_ACT_GRAD),
```

Comment:
`@not_implemented` is designed for exactly this. But seems not to be GPU-friendly, sadly.

Reply:
With NaN instead, the example from https://github.com/FluxML/NNlib.jl/issues/386 runs.

Reply:
No, in fact it runs forwards, but not backwards:
```
julia> l, b = Flux.pullback(() -> gradient_penalty(cm₁, cx), params(cm₁)) # Fails to compile
(0.00047222938f0, Zygote.var"#94#95"{Params, typeof(∂(#45)), Zygote.Context}(Params([Float32[-0.015067626;;], Float32[0.0]]), ∂(#45), Zygote.Context(IdDict{Any, Any}(Zygote.Stack{Any}(0, Any[]) => Base.RefValue{Any}((idx = nothing, data = nothing)), Zygote.Stack{Any}(0, Any[]) => Base.RefValue{Any}((idx = nothing, data = nothing)), Zygote.Stack{Any}(0, Any[]) => Base.RefValue{Any}((idx = nothing, data = nothing)), Zygote.Stack{Any}(0, Any[]) => Base.RefValue{Any}((idx = nothing, data = nothing)), Zygote.Stack{Any}(0, Any[]) => Base.RefValue{Any}((idx = nothing, data = nothing)), Zygote.Stack{Any}(0, Any[]) => Base.RefValue{Any}((idx = nothing, data = nothing)), Zygote.Stack{Any}(1, Any[Zygote.ZBack{ChainRules.var"#===_pullback#81"}(ChainRules.var"#===_pullback#81"())]) => Base.RefValue{Any}((idx = nothing, data = nothing)), Zygote.Stack{Any}(1, Any[Zygote.ZBack{ChainRules.var"#===_pullback#81"}(ChainRules.var"#===_pullback#81"())]) => Base.RefValue{Any}((idx = nothing, data = nothing)), Zygote.Stack{Any}(0, Any[]) => Base.RefValue{Any}((idx = nothing, data = nothing)), Zygote.Stack{Any}(0, Any[]) => Base.RefValue{Any}((idx = nothing, data = nothing))…))))

julia> b(1f0)
ERROR: Mutating arrays is not supported -- called copyto!(::CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}, _...)
Stacktrace:
  [1] error(s::String)
    @ Base ./error.jl:33
  [2] (::Zygote.var"#446#447"{CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}})(#unused#::CuArray{Float32, 2, CUDA.Mem.DeviceBuffer})
    @ Zygote ~/.julia/packages/Zygote/FPUm3/src/lib/array.jl:74
  [3] (::Zygote.var"#2351#back#448"{Zygote.var"#446#447"{CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}}})(Δ::CuArray{Float32, 2, CUDA.Mem.DeviceBuffer})
    @ Zygote ~/.julia/packages/ZygoteRules/AIbCs/src/adjoint.jl:67
  [4] Pullback
    @ ./broadcast.jl:871 [inlined]
  [5] Pullback
    @ ./broadcast.jl:868 [inlined]
  [6] Pullback
    @ ./broadcast.jl:864 [inlined]
  [7] Pullback
    @ ~/.julia/packages/Zygote/FPUm3/src/lib/broadcast.jl:275 [inlined]
  [8] (::typeof(∂(λ)))(Δ::Tuple{CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}})
    @ Zygote ./compiler/interface2.jl:0
  [9] Pullback
    @ ~/.julia/packages/ZygoteRules/AIbCs/src/adjoint.jl:67 [inlined]
 [10] (::typeof(∂(λ)))(Δ::Tuple{Nothing, CuArray{Float32, 2, CUDA.Mem.DeviceBuffer}})
    @ Zygote ./compiler/interface2.jl:0
 [11] Pullback
    @ ./REPL[27]:2 [inlined]
 [12] (::typeof(∂(λ)))(Δ::Nothing)
    @ Zygote ./compiler/interface2.jl:0
 [13] Pullback
    @ ~/.julia/packages/Zygote/FPUm3/src/compiler/interface.jl:357 [inlined]
 [14] (::typeof(∂(λ)))(Δ::Nothing)
    @ Zygote ./compiler/interface2.jl:0
 [15] Pullback
    @ ./REPL[27]:3 [inlined]
 [16] (::typeof(∂(gradient_penalty)))(Δ::Float32)
    @ Zygote ./compiler/interface2.jl:0
 [17] Pullback
    @ ./REPL[36]:1 [inlined]
 [18] (::typeof(∂(#45)))(Δ::Float32)
    @ Zygote ./compiler/interface2.jl:0
 [19] (::Zygote.var"#94#95"{Params, typeof(∂(#45)), Zygote.Context})(Δ::Float32)
    @ Zygote ~/.julia/packages/Zygote/FPUm3/src/compiler/interface.jl:357
 [20] top-level scope
    @ REPL[37]:1
 [21] top-level scope
    @ ~/.julia/packages/CUDA/KnJGx/src/initialization.jl:52
```